eslint.config.js

```diff
@@ -0,0 +1,732 @@
+  'no-fallthrough': 1, // Was 2
+  'no-func-assign': 1, // Was 2
+  // "no-import-assign": 2,
+  'no-inner-declarations': 0, // Was 2
+  // "no-invalid-regexp": 2,
+  'no-irregular-whitespace': [
+    1, // Was 2
+    {
+      // skipStrings: true,
+      // skipComments: false,
+      // skipRegExps: false,
+      skipTemplates: true
+    }
+  ],
+  'no-loss-of-precision': 1, // Was 2
+  'no-misleading-character-class': 1, // Was 2
+  // "no-new-symbol": 2,
+  // "no-obj-calls": 2,
+  'no-promise-executor-return': 1,
+  // "no-prototype-builtins": 2,
+  'no-self-assign': 1, // Was 2
+  'no-self-compare': 1,
+  'no-setter-return': 1, // Was 2
+  'no-sparse-arrays': 1, // Was 2
+  'no-template-curly-in-string': 1,
+  // "no-this-before-super": 2,
+  'no-undef': [2, { 'typeof': true }],
+  'no-unexpected-multiline': 1, // Was 2
+  'no-unmodified-loop-condition': 1,
+  'no-unreachable': 1, // Was 2
+  'no-unreachable-loop': 1,
+  'no-unsafe-finally': 1, // Was 2
+  'no-unsafe-negation': [
+    1, // Was 2
+    { enforceForOrderingRelations: true }
+  ],
+  'no-unsafe-optional-chaining': [2, { disallowArithmeticOperators: true }],
+  'no-unused-private-class-members': 1,
+  'no-unused-vars': [
+    1, // Was 2
+    {
+      // vars: "all",
+      // args: "after-used",
+      // ignoreRestSiblings: false,
+      argsIgnorePattern: '^_',
+      caughtErrors: 'all', // Was "none"
+      caughtErrorsIgnorePattern: '^_'
+    }
+  ],
+  'no-use-before-define': [
+    1,
+    {
+      functions: false
+      // classes: true,
+      // variables: true
+    }
+  ],
+  'no-useless-backreference': 1,
+  'require-await': 1,
+  'require-atomic-updates': 1,
+  'use-isnan': [
+    1, // Was 2
+    {
+      // enforceForSwitchCase: true,
+      enforceForIndexOf: true
+    }
+  ],
+  'valid-typeof': [
+    1, // Was 2
+    { requireStringLiterals: true }
+  ]
+};
+
+const suggestions = {
+  'accessor-pairs': 1,
+  'arrow-body-style': 1,
+  'block-scoped-var': 1,
+  'camelcase': 1,
+  // "capitalized-comments": 0, // Allow commented code
+  // "class-methods-use-this": 0,
+  'complexity': 1,
+  'consistent-return': 1,
+  'consistent-this': 1,
+  'curly': [
+    1,
+    'multi-line' // Was "all"
+  ],
+  // "default-case": 0,
+  'default-case-last': 1,
+  'default-param-last': 1,
+  'dot-notation': 1,
+  'eqeqeq': 1,
+  'func-name-matching': [
+    1,
+    'always', // Same
+    {
+      considerPropertyDescriptor: true
+      // includeCommonJSModuleExports: false
+    }
+  ],
+  // "func-names": 0,
+  'func-style': 0, // Was 1
+  'grouped-accessor-pairs': [
+    1,
+    'getBeforeSet' // Was "anyOrder"
+  ],
+  // "guard-for-in": 0,
+  // "id-denylist": 0,
+  // "id-length": 0,
+  // "id-match": 0,
+  // "init-declarations": 0,
+  // "max-classes-per-file": 0,
+  // "max-depth": 0,
+  // "max-lines": 0,
+  // "max-lines-per-function": 0,
+  // "max-nested-callbacks": 0,
+  // "max-params": 0,
+  // "max-statements": 0,
+  // "multiline-comment-style": 0,
```

Comment:
These came from overrides from the old (now removed) AirBnB preset right? Can we just remove them? In fact I think might be best to purge all the overrides and start from a more modern preset like `@typescript-eslint/recommended` (or even better, `@typescript-eslint/recommended-type-checked`)

Reply:
I'll go through everything and see if I can streamline the configuration